lib/rules/semi-style.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Rule to enforce location of semicolons.
+ * @author Toru Nagashima
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const SELECTOR = `:matches(${
+    [
+        "BreakStatement", "ContinueStatement", "DebuggerStatement",
+        "EmptyStatement", "ExportAllDeclaration", "ExportDefaultDeclaration",
+        "ExportNamedDeclaration", "ExpressionStatement", "ImportDeclaration",
+        "ReturnStatement", "ThrowStatement", "VariableDeclaration"
+    ].join(",")
+})`;
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce location of semicolons",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        schema: [{
+            anyOf: [
+                { enum: ["last", "first"] },
+                {
+                    type: "object",
+                    properties: {
+                        statements: { enum: ["last", "first"] },
+                        forLoopHead: { enum: ["last", "first"] }
+                    },
+                    additionalProperty: false
+                }
+            ]
+        }],
+        fixable: "whitespace"
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+        const options = context.options[0] || "last";
+        const statementLoc = typeof options === "string" ? options : (options.statements || "last");
```

Comment:
Nitpick: Can you rename this variables to something like `statementOption`? When I saw `loc` later on in the file, I thought it was referring to a `{ line: number, column: number }` object.